src/ser.rs

```diff
@@ -1,15 +1,54 @@
 use std::marker::PhantomData;
 
-use pyo3::types::{PyAnyMethods, PyDict, PyList, PyMapping, PySequence, PyString, PyTuple};
+use pyo3::types::{
+    PyAnyMethods, PyDict, PyDictMethods, PyList, PyMapping, PySequence, PyString, PyTuple,
+    PyTupleMethods,
+};
 use pyo3::{Bound, IntoPy, PyAny, PyResult, Python, ToPyObject};
 use serde::{ser, Serialize};
 
 use crate::error::{PythonizeError, Result};
 
 /// Trait for types which can represent a Python mapping
-pub trait PythonizeDictType {
-    /// Constructor
-    fn create_mapping(py: Python) -> PyResult<Bound<PyMapping>>;
+pub trait PythonizeMappingType {
+    /// Builder type for Python mappings
+    type Builder<'py>;
+
+    /// Create a builder for a Python mapping
+    fn builder(py: Python, len: Option<usize>) -> PyResult<Self::Builder<'_>>;
```

Comment:
To avoid the GAT we'll have to move the lifetime onto the trait.

Maybe leave a TODO to clean up in future when we bump MSRV?

```suggestion
pub trait PythonizeMappingType<'py> {
    /// Builder type for Python mappings
    type Builder;

    /// Create a builder for a Python mapping
    fn builder(py: Python<'py>, len: Option<usize>) -> PyResult<Self::Builder>;
```

Reply:
Done